tools/releasestaging

```diff
@@ -6,21 +6,101 @@ dry=echo
 [ "$DRYRUN" = 0 ] && dry=
 
 FORCE=1 $dry tools/cleanuprepo $sloreleasing
-#pkgs=`osc ls $slo:Staging`
 shopt -s nullglob
-pkgs=out/pending/*
+pkgs=(out/pending/*)
 releasedpkgs=""
-for pkg in $pkgs ; do
+cache_dir="tmp/cache"
+
+# Ensure the cache directory exists
+mkdir -p $cache_dir
+
+# Function to download binaries once and cache them
+function osc_getbinaries_once {
+    repo=$1
+    pkg=$2
+    dest=$3
+    cachedir=$cache_dir/$pkg
+
+    # If binaries are already cached, skip download
+    if [ -d "$cachedir" ]; then
+        echo "Using cached binaries for $pkg"
+        cp -r $cachedir/* $dest/
+    else
+        echo "Downloading binaries for $pkg"
+        osc ls -b $repo $pkg standard x86_64 | grep -q .
+        ret=$?
+        if [[ $ret = 0 ]] ; then
```

Comment:
Instead of assigning a variable and checking $? you can just `if  osc ls -b $repo $pkg standard x86_64 | grep -q .`